### PR TITLE
Homework 6

### PR DESCRIPTION
**Pull request by povilassta**
Authentication and routing

### CODE REVIEW

File: client/src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+import { AuthService } from './auth/services/auth.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.sass']
+  styleUrls: ['./app.component.sass'],
 })
 export class AppComponent {
   title = 'bugetify';
```
Comment (AnastasiaHarost) [inline client/src/app/app.component.ts:11]: please, add modifiers public/private/protected before every variable and function

File: client/src/app/app.component.ts [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+
+  constructor(private authService: AuthService, private router: Router) {}
+
+  get isLoggedIn() {
+    return this.authService.isLoggedIn();
+  }
+
+  logout() {
```
Comment (AnastasiaHarost) [inline client/src/app/app.component.ts:19]: please, add modifiers public/private/protected before every function and add return type of the function

File: client/src/app/auth/auth-form/auth-form.component.sass [hunk 3]
```diff
@@ -0,0 +1,27 @@
+.container
+    display: flex
```
Comment (AnastasiaHarost) [inline client/src/app/auth/auth-form/auth-form.component.sass:2]: MINOR add, please,   ;   in the end of lines

File: client/src/app/auth/auth-form/auth-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { Router } from '@angular/router';
+import { AuthService } from '../services/auth.service';
+
+@Component({
+  selector: 'app-auth-form',
+  templateUrl: './auth-form.component.html',
+  styleUrls: ['./auth-form.component.sass'],
+})
+export class AuthFormComponent {
+  hide = true;
```
Comment (AnastasiaHarost) [inline client/src/app/auth/auth-form/auth-form.component.ts:12]: modifiers

File: client/src/app/auth/auth-form/auth-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  loginForm: FormGroup = new FormGroup({
+    email: new FormControl('', [Validators.required]),
+    password: new FormControl('', [Validators.required]),
+  });
+  constructor(private authService: AuthService, private router: Router) {}
+
+  onSubmit() {
```
Comment (AnastasiaHarost) [inline client/src/app/auth/auth-form/auth-form.component.ts:19]: modifier and return type

File: client/src/app/auth/auth-form/auth-form.component.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    const { email, password } = this.loginForm.value;
+    this.authService.login(email, password).subscribe((data) => {
+      console.log('successful login');
```
Comment (AnastasiaHarost) [inline client/src/app/auth/auth-form/auth-form.component.ts:22]: remove console.log

File: client/src/app/auth/services/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import { tap } from 'rxjs';
+import * as moment from 'moment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor(private http: HttpClient) {}
+
+  login(email: string, password: string) {
```
Comment (AnastasiaHarost) [inline client/src/app/auth/services/auth.service.ts:12]: modifiers, return type for all functions

File: client/src/app/main/main/main.component.html [hunk 8]
```diff
@@ -0,0 +1 @@
+<p>main works!</p>
```
Comment (AnastasiaHarost) [inline client/src/app/main/main/main.component.html:1]: should be empty

File: client/src/app/auth/services/auth.service.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    const expiration = localStorage.getItem('expiresAt');
+    const expiresAt = JSON.parse(localStorage.getItem('expiresAt') || '{}');
+    return moment(expiresAt);
+  }
+
+  private setSession(res: any) {
+    const expiresAt = moment().add(1, 'h');
+    localStorage.setItem('token', res.token);
+    localStorage.setItem('expiresAt', JSON.stringify(expiresAt.valueOf()));
+    console.log(JSON.stringify(expiresAt.valueOf()));
```
Comment (AnastasiaHarost) [inline client/src/app/auth/services/auth.service.ts:37]: no console.logs

File: client/src/app/auth/auth-form/auth-form.component.sass (hunk 3 above)
Comment (povilassta) [reply client/src/app/auth/auth-form/auth-form.component.sass:2]: SASS doesn't use semicolumns at the end of lines

File: client/src/app/auth/services/auth.service.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    localStorage.removeItem('token');
+  }
+
+  private getExpiration(): moment.Moment {
+    const expiration = localStorage.getItem('expiresAt');
+    const expiresAt = JSON.parse(localStorage.getItem('expiresAt') || '{}');
+    return moment(expiresAt);
+  }
+
+  private setSession(res: any): void {
```
Comment (AnastasiaHarost) [inline client/src/app/auth/services/auth.service.ts:33]: avoid using any

File: client/src/app/auth/auth-form/auth-form.component.sass (hunk 3 above)
Comment (povilassta) [reply client/src/app/auth/auth-form/auth-form.component.sass:2]: So i should switch all style file extensions to scss?

File: client/src/app/auth/auth-form/auth-form.component.ts [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+import { Router } from '@angular/router';
+import { AuthService } from '../services/auth.service';
+
+@Component({
+  selector: 'app-auth-form',
+  templateUrl: './auth-form.component.html',
+  styleUrls: ['./auth-form.component.sass'],
+})
+export class AuthFormComponent {
+  public hide = true;
```
Comment (AnastasiaHarost) [inline client/src/app/auth/auth-form/auth-form.component.ts:12]: add type, please

File: client/src/app/auth/auth-form/auth-form.component.sass (hunk 3 above)
Comment (AnastasiaHarost) [reply client/src/app/auth/auth-form/auth-form.component.sass:2]: no, let it stay like that

File: client/src/app/auth/services/auth.service.ts [hunk 12]
```diff
@@ -30,7 +31,8 @@ export class AuthService {
     return moment(expiresAt);
   }
 
-  private setSession(res: any): void {
+  private setSession(res: LoginType): void {
+    console.log(res);
```
Comment (AnastasiaHarost) [inline client/src/app/auth/services/auth.service.ts:35]: remove console.log

**Review by AnastasiaHarost — CHANGES_REQUESTED** [review]

**Pull request merged**